akka-http-testkit/src/test/scala/akka/http/scaladsl/testkit/ScalatestRouteTestSpec.scala

```diff
@@ -38,6 +45,46 @@ class ScalatestRouteTestSpec extends FreeSpec with Matchers with ScalatestRouteT
       }
     }
 
+    //tests that run the route without check() - these complete immediately even if the future doesn't
+    "a route that immediately responds in Future.successful - without check {}" in {
+      stringResponse(Get("http://localhost/futures/quick") ~> Route.seal(Routes.route)) shouldEqual "done-quick"
+    }
+
+    "a route with a quick async boundary - without check {}" in {
+      stringResponse(Get("http://localhost/futures/buggy") ~> Route.seal(Routes.route)) shouldEqual "done-buggy"
+    }
+
+    "a long running future - without check {}" in {
+      stringResponse(Get("http://localhost/futures/long") ~> Route.seal(Routes.route)) shouldEqual "done-long"
+    }
+
+    def stringResponse(result: RouteTestResult): String = {
+      Await.result(Unmarshal(result.response).to[String], 5.seconds)
```

Comment:
This is not a valid way to use `result.response`. You need to call `result.awaitResult` before accessing any of the other fields (which is done by `check`). Adding `result.awaitResult` before this line fixes one of the tests.

Reply:
Yes. But the users can't do that (and I wanted these tests to show what the problem for users is). I'm fine with getting rid of this way to get the response, and getting the `check` examples to work :)